src/test/it/jenkins.yml

```diff
@@ -0,0 +1,36 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    iam.amazonaws.com/role: <IAM_ROLE>
+  generateName: <NAME>-
+  labels:
+    name: jnlp
```

Comment:
The filename `jenkins.yml` implies that this is a pod for the master, yet you seem to be defining an agent here?

Reply:
this file is used in the pipeline library to provision a pod